docs/book/starter-guide/pipelines/fetching-pipelines.md

```diff
@@ -253,6 +280,19 @@ def some_step() -> Output(output_name=int):
 ```
 {% endhint %}
 
+### Output Artifact Metadata
+
+All output artifacts saved through ZenML will automatically have certain
+datatype-specific metadata saved with them. NumPy Arrays, for instance, always
+have their storage size, `shape`, `dtype`, and some statistical properties
```

Comment:
Do we have a reference to API docs or just the code where we list exactly what properties are saved with them? Or are they embedded in each integration separately? Maybe give an example for where someone would go to find these properties for the NumPy arrays that you specify, just so people know where to look to find this out, aside from just running the code.

Reply:
Yeah the contents of this dict change depending on the artifact. I guess you could look into the code of the respective materializer to figure it out but not sure that's what we want to lead users to do. IMO programmatically getting the metadata and just seeing what keys are in there is the best way to find out at the moment. Also, these properties are not designed as something that users will ever need to manually fetch since they will soon automatically be displayed in the dashboard next to the info of each run / step. I guess including screenshots of that in the docs will make it clearer once we have this.